apps/OpenSpace/ext/launcher/src/sgctedit/orientation.cpp

```diff
@@ -0,0 +1,67 @@
+ *                                                                                       *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy of this  *
+ * software and associated documentation files (the "Software"), to deal in the Software *
+ * without restriction, including without limitation the rights to use, copy, modify,    *
+ * merge, publish, distribute, sublicense, and/or sell copies of the Software, and to    *
+ * permit persons to whom the Software is furnished to do so, subject to the following   *
+ * conditions:                                                                           *
+ *                                                                                       *
+ * The above copyright notice and this permission notice shall be included in all copies *
+ * or substantial portions of the Software.                                              *
+ *                                                                                       *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR IMPLIED,   *
+ * INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY, FITNESS FOR A         *
+ * PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT    *
+ * HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF  *
+ * CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE  *
+ * OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.                                         *
+ ****************************************************************************************/
+
+#include "sgctedit/display.h"
+#include "sgctedit/orientation.h"
+#include "sgctedit/orientationdialog.h"
+
+Orientation::Orientation()
+{
+    _orientationDialog = new OrientationDialog(_orientationValue, this);
+    _layoutOrientationFull = new QHBoxLayout();
+    {
+        QVBoxLayout* layoutOrientationControls = new QVBoxLayout();
+        QPushButton* orientationButton = new QPushButton("Global Orientation");
+        _checkBoxVsync = new QCheckBox("VSync All Windows", this);
+        layoutOrientationControls->addWidget(_checkBoxVsync);
+        layoutOrientationControls->addWidget(orientationButton);
+        _layoutOrientationFull->addStretch(1);
+        _layoutOrientationFull->addLayout(layoutOrientationControls);
+        _layoutOrientationFull->addStretch(1);
+        connect(orientationButton, SIGNAL(released()), this,
+                SLOT(orientationDialog()));
+    }
+}
+
+void Orientation::addButtonToLayout(QVBoxLayout* parentLayout) {
+    parentLayout->addLayout(_layoutOrientationFull);
+}
+
+void Orientation::orientationDialog() {
+    _orientationDialog->exec();
+}
+
+sgct::quat Orientation::orientationValue() {
+    return _orientationValue;
+}
+
+bool Orientation::vsyncValue() {
+    return (_checkBoxVsync->checkState() == Qt::Checked);
+}
+
+Orientation::~Orientation()
```

Comment:
If moving the `_orientationDialog` to be not a pointer, the destructor is no longer necessary since the layout will be automatically destroyed by Qt when the parent layout is destroyed